riscv/debug_module.cc

```diff
@@ -487,7 +463,7 @@ bool debug_module_t::dmi_read(unsigned address, uint32_t *value)
           unsigned base = hawindowsel * 32;
           for (unsigned i = 0; i < 32; i++) {
             unsigned n = base + i;
-            if (n < nprocs && hart_array_mask[n]) {
+            if (n < hart_array_mask.size() && hart_array_mask[n]) {
```

Comment:
n here is a debug hart index, not a hart ID. So can't directly index into hart_array_mask.